app/models/manageiq/providers/kubevirt/infra_manager/connection.rb

```diff
@@ -17,18 +17,31 @@ class ManageIQ::Providers::Kubevirt::InfraManager::Connection
   def initialize(opts = {})
     require 'fog/kubevirt'
     # create fog based connection
+    @opts = opts
     @conn = Fog::Kubevirt::Compute.new(
-      :kubevirt_hostname  => opts[:host],
-      :kubevirt_port      => opts[:port],
-      :kubevirt_token     => opts[:token],
-      :kubevirt_namespace => opts[:namespace] || 'default',
+      :kubevirt_hostname  => @opts[:host],
+      :kubevirt_port      => @opts[:port],
+      :kubevirt_token     => @opts[:token],
+      :kubevirt_namespace => @opts[:namespace] || 'default',
       :kubevirt_log       => $log
     )
 
     # Nothing else is done here, as this method should never throw an exception, even if the
     # credentials are wrong.
   end
 
+  def namespaced_client(namespace)
+    require 'fog/kubevirt'
+
+    namespaced_client = Fog::Kubevirt::Compute.new(
+      :kubevirt_hostname  => @opts[:host],
+      :kubevirt_port      => @opts[:port],
+      :kubevirt_token     => @opts[:token],
+      :kubevirt_namespace => namespace,
+      :kubevirt_log       => $log
+    )
+  end
```

Comment:
`@conn` is the client connection that uses the catch-all namespace `""` and `namespaced_client` is the client connection targeting a specific namespace. Not a fan of this but didn't see another way until we proceed with replacing the fog client with `kubeclient`